From c5bc0073966ee032d86b9a9f4c0b00f7735ca958 Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sujeet.baranwal@cavium.com>
Date: Tue, 27 Nov 2018 04:01:12 +0300
Subject: [PATCH 0409/1921] mmc: octeontx2: use device tree entries

The characterstic of the device shall be determined
based on the parameters mentioned in the device tree
instead of hardcoding in the driver.

Change-Id: Icbadb0d9332397423d97f1e2459bdd961bcefca0
Signed-off-by: Sujeet Baranwal <sujeet.baranwal@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/8538
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-by: Sujeet Kumar Baranwal <Sujeet.Baranwal@cavium.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/26904
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/mmc/host/cavium.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/mmc/host/cavium.c b/drivers/mmc/host/cavium.c
index c390c00d4b55..ebd7d0cb453c 100644
--- a/drivers/mmc/host/cavium.c
+++ b/drivers/mmc/host/cavium.c
@@ -1210,15 +1210,18 @@ int cvm_mmc_of_slot_probe(struct device *dev, struct cvm_mmc_host *host)
 	mmc->ops = &cvm_mmc_ops;
 
 	/*
-	 * We only have a 3.3v supply, we cannot support any
-	 * of the UHS modes. We do support the high speed DDR
-	 * modes up to 52MHz.
+	 * We only have a 3.3v supply for slots, we cannot
+	 * support any of the UHS modes. We do support the
+	 * high speed DDR modes up to 52MHz.
 	 *
 	 * Disable bounce buffers for max_segs = 1
 	 */
+
 	mmc->caps |= MMC_CAP_MMC_HIGHSPEED | MMC_CAP_SD_HIGHSPEED |
-		     MMC_CAP_ERASE | MMC_CAP_CMD23 | MMC_CAP_POWER_OFF_CARD |
-		     MMC_CAP_3_3V_DDR;
+		MMC_CAP_ERASE | MMC_CAP_CMD23 | MMC_CAP_POWER_OFF_CARD;
+
+	if (!is_mmc_otx2(host))
+		mmc->caps |= MMC_CAP_3_3V_DDR;
 
 	if (host->use_sg)
 		mmc->max_segs = 16;
-- 
2.31.1

